suite-native/module-authenticity-checks/src/screens/DeviceCompromisedModalScreen.tsx

```diff
@@ -1,19 +1,29 @@
-import { useSelector } from 'react-redux';
-
-import { selectIsEntropyCheckEnabledAndFailed } from '@suite-native/device';
+import { UnreachableCaseError } from '@suite-common/suite-utils';
+import { RootStackParamList, RootStackRoutes, StackProps } from '@suite-native/navigation';
 
+import { DeviceAuthenticityCheckFailModalContent } from '../components/DeviceAuthenticityCheckFailModalContent';
 import { EntropyCheckFailModalContent } from '../components/EntropyCheckFailModalContent';
 import { FirmwareAuthenticityCheckFailModalContent } from '../components/FirmwareAuthenticityCheckFailModalContent';
 
 /**
- * The very similar modal can be displayed for entropy check failure or FW authenticity check failure
+ * Modal can be displayed for:
+ * - entropy check failure
+ * - FW authenticity check failure
+ * - device authenticity check failure
  */
-export const DeviceCompromisedModalScreen = () => {
-    const isEntropyCheckEnabledAndFailed = useSelector(selectIsEntropyCheckEnabledAndFailed);
+export const DeviceCompromisedModalScreen = ({
```

Comment:
What's the added value of having `Modal` in the name? There are other screens that have similar behaviour but are not named like this. I'd drop `Modal` from all the names in this PR.

Reply:
From my perspective `Modal` should not be in the name of this screen, but the diff to remove it is quite big, so I would rather do this as pure refactoring PR as followup:
[remove-modal-name.diff.txt](https://github.com/user-attachments/files/20160966/remove-modal-name.diff.txt)

Reply:
Follow-up OK with me.